syntax = "proto3";

package battleshipservice;

// Board Service
service BoardService {
  rpc BattleShip (CreateBoardRequest) returns (CreateBoardResponse);
}

// representing 2d array: https://stackoverflow.com/a/15179330
message Board {
  int32 rows = 1;
  int32 cols = 2;
  repeated double data = 3 [packed=true];
}

message CreateBoardRequest{
  repeated Board board = 1;
}

message CreateBoardResponse{}

message PlaceShipVertRequest{

}

message PlaceShipVertResponse{}

message PlaceShipHorRequest{

}

message PlaceShipHorResponse{}

message IsValidPlacementRequest{

}

message IsValidPlacementResponse{
  
}

// Game Service

// service UnaryService {
//   rpc BattleShip (BattleShipRequest) returns (BattleShipResponse) {}
// }

// service ServerStreamingService {
//   rpc BattleShip (BattleShipRequest) returns (stream BattleShipStreamingResponse) {}
// }

// message BattleShipRequest {
//   string message = 1;
// }

// message BattleShipResponse {
//   string message = 1;
// }

// message BattleShipStreamingResponse {
//   message BattleShipStreamEvent {
//     string message = 1;
//   }

//   repeated BattleShipStreamEvent events = 1;
// }